pkg/k8shandler/migrations/migrations.go

```diff
@@ -0,0 +1,62 @@
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/openshift/elasticsearch-operator/pkg/elasticsearch"
+	"github.com/openshift/elasticsearch-operator/pkg/utils"
+	"github.com/sirupsen/logrus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type MigrationRequest interface {
+	RunKibanaMigrations() error
+	RunElasticsearchMigrations() error
+}
+
+func NewMigrationRequest(client client.Client, esClient elasticsearch.Client) MigrationRequest {
+	return &migrationRequest{
+		client:   client,
+		esClient: esClient,
+	}
+}
+
+type migrationRequest struct {
+	client   client.Client
+	esClient elasticsearch.Client
+}
+
+func (mr *migrationRequest) RunKibanaMigrations() error {
+	logrus.Debugf("running Kibana migrations")
+	if status, _ := mr.esClient.GetClusterHealthStatus(); status != "green" {
```

Comment:
i would think we actually just need to ensure the .kibana index is `green` not the whole cluster.
also, pertaining to cluster upgrades we allow `yellow` as well